core/src/test/java/io/questdb/test/griffin/engine/functions/groupby/ApproxCountDistinctIPv4GroupByFunctionFactoryTest.java

```diff
@@ -0,0 +1,232 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2023 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.test.griffin.engine.functions.groupby;
+
+import io.questdb.test.AbstractCairoTest;
+import org.junit.Test;
+
+public class ApproxCountDistinctIPv4GroupByFunctionFactoryTest extends AbstractCairoTest {
```

Comment:
None of the tests are multi-threaded, i.e. the only test which tests merge is `HyperLogLogTest`. Would be great to add a few tests to `ParallelGroupByFuzzTest`, with both `approx_count_distinct()` and `count_distinct()` as aggregate functions (that's to make the test results easier to verify by a human).

Reply:
Done: 3ad1e59ac33ce7248b84281faf5c037d6e80cb79

Reply:
All of the added tests seem to use sparse representation. Is that correct? If so, would be great to add a test that would hit dense representation.

Reply:
Done: 072dfb28ef4a88c147e83fc3c33b36f537126a7d